bytecomp/lambda.mli

```diff
@@ -41,6 +41,7 @@ type tag_info =
   | Blk_extension
   | Blk_na
   | Blk_some
+  | Blk_some_special (* ['a option] where ['a] can not inhabit a non-like value *)
```

Comment:
How about `Blk_some_shallow` or `Blk_some_not_nested`?
I guess the comment should say "a' will not be instantiated with an option type."